src/client.rs

```diff
@@ -187,6 +306,31 @@ pub struct Collection {
 /// ### Custom
 /// See [`Verification`] for implementing a custom signature verifier.
 ///
+/// ## Attachments
+///
+/// When built with the `attachments` feature, attachments associated with records can be downloaded.
+///
+/// ```rust
+/// # use remote_settings_client::Client;
+/// # #[tokio::main]
+/// # #[cfg(feature = "attachments")]
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let client = Client::builder()
+///   .collection_name("cid")
+///   .build()
+///   .unwrap();
+///
+/// // Attachment operations store cache data on records, so they need mutable references.
+/// let mut records = client.get().await?;
+/// let attachment_metadata = records[0].attachment_metadata()?;
```

Comment:
Maybe, as an example of reading metadata, we could add a log print here announcing the size in bytes to be downloaded